ProcessMaker/processmaker
# Unattended installer

## Changes
- Utilizes the `-n` flag to allow users to install ProcessMaker noninteractively via CLI
- Adds dozens of flags to enable users to set various configuration options
- Adds a `-p` (pretend) flag to enable users to see how their configuration options will impact the .env file
- Adds testing to ensure provided flags properly set configuration values
- Removes faker placeholder data from the generated admin user

## Testing Instructions
- Run `art processmaker:install` to ensure the interactive installer still works as intended
- Run `art processmaker:install -h` to see a list of the new flags
- Run `art processmaker:install -n` to ensure that a non-interactive installation will fail with no configuration options
- Attempt a fully unattended installation by running:
`art processmaker:install -n --url=https://processmaker.local.processmaker.com --username=admin --password=admin123 --email=admin@processmaker.com --db-host=127.0.0.1 --db-port=3306 --db-name=processmaker --db-username=homestead --db-password=secret --data-driver=none --first-name=Default --last-name=Administrator`

## Screenshot
![Screen Shot 2020-01-21 at 12 06 50 PM](https://user-images.githubusercontent.com/867714/72839154-985f9680-3c46-11ea-9dbc-27f24360d574.png)

Closes #2736.
